Sources/FTPropertyWrappers/Keychain - Support/KeychainCoding.swift

```diff
@@ -0,0 +1,371 @@
+
+    private let encoder: PropertyListEncoder = {
+        let newEncoder = PropertyListEncoder()
+        newEncoder.outputFormat = .binary
+        return newEncoder
+    }()
+
+    private let decoder: PropertyListDecoder = PropertyListDecoder()
+
+    func encode<T: Encodable>(_ value: T) throws -> Data {
+        switch value {
+        case let value as Int:
+            return try encode(value)
+        case let value as Int8:
+            return try encode(value)
+        case let value as Int16:
+            return try encode(value)
+        case let value as Int32:
+            return try encode(value)
+        case let value as Int64:
+            return try encode(value)
+        case let value as UInt:
+            return try encode(value)
+        case let value as UInt8:
+            return try encode(value)
+        case let value as UInt16:
+            return try encode(value)
+        case let value as UInt32:
+            return try encode(value)
+        case let value as UInt64:
+            return try encode(value)
+        case is Float:
+            fatalError("Encoding root type Float not supported!")
+        case is Float80:
+            fatalError("Encoding root type Float80 not supported!")
+        case is Double:
+            fatalError("Encoding root type Double not supported!")
+        case let value as Bool:
+            return try encode(value)
+        case let value as String:
+            return try encode(value)
+        case let value as URL:
+            return try encode(value)
+        case let value as Data:
+            return try encode(value)
+        default:
+            return try encoder.encode(value)
+        }
+    }
+
+    func decode<T: Decodable>(_ type: T.Type, from data: Data) throws -> T {
+        switch type {
+        case is Int.Type:
+            return (try decode(from: data) as Int) as! T
+        case is Int8.Type:
+            return (try decode(from: data) as Int8) as! T
+        case is Int16.Type:
+            return (try decode(from: data) as Int16) as! T
+        case is Int32.Type:
+            return (try decode(from: data) as Int32) as! T
+        case is Int64.Type:
+            return (try decode(from: data) as Int64) as! T
+        case is UInt.Type:
+            return (try decode(from: data) as UInt) as! T
+        case is UInt8.Type:
+            return (try decode(from: data) as UInt8) as! T
+        case is UInt16.Type:
+            return (try decode(from: data) as UInt16) as! T
+        case is UInt32.Type:
+            return (try decode(from: data) as UInt32) as! T
+        case is UInt64.Type:
+            return (try decode(from: data) as UInt64) as! T
+        case is Float.Type:
+            fatalError("Encoding root type Float not supported!")
+        case is Float80.Type:
+            fatalError("Encoding root type Float80 not supported!")
+        case is Double.Type:
+            fatalError("Encoding root type Double not supported!")
```

Comment:
You are mixing fatal errors with exceptions. I would unify the error handling system.

```suggestion
            throws DecodingError.dataCorrupted(
                DecodingError.Context(codingPath: [], debugDescription: "Encoding root type Double not supported!", underlyingError: nil)
            )
```